packages/core/src/editor/BlockNoteEditor.ts

```diff
@@ -1073,18 +1076,28 @@ export class BlockNoteEditor<
     targetBlock: BlockIdentifier,
     placement: "start" | "end" = "start"
   ) {
-    setTextCursorPosition(this, targetBlock, placement);
+    return this.transact((tr) =>
```

Comment:
why are some of the methods like this one wrapped in `transact`?

Reply:
`transact` is convenient in that you don't have to create a transaction & then dispatch it. It also is clear that it would work within a blocknote transaction.

If everything were to use this pattern, then you wouldn't even need to have `editor.transaction` hold the `activeTransaction` state, it would just be passed around through the `transact` calls. So, it ends up mapping more closely to what actually happens

Reply:
Ah that makes sense! I was looking at it from the angle of "only call `transact` if you're doing multiple BlockNote API calls in a method.

But you're right, we could also use it for convenience + make the methods consistent. I suppose in that case we need to do it for all usages of `dispatch`, and perhaps even make `dispatch` private?

Reply:
Good call on standardizing it, will refactor to that pattern. I think it will come out nice